checked_message_bench_test.go

```diff
@@ -0,0 +1,96 @@
+package zap_test
+
+import (
+	"testing"
+
+	"github.com/uber-go/atomic"
+	"github.com/uber-go/zap"
+)
+
+func BenchmarkCheckedMessage_Chain(b *testing.B) {
+	infoLog := zap.New(
+		zap.NullEncoder(),
+		zap.InfoLevel,
+		zap.DiscardOutput,
+	)
+	errorLog := zap.New(
+		zap.NullEncoder(),
+		zap.ErrorLevel,
+		zap.DiscardOutput,
+	)
+
+	data := []struct {
+		lvl zap.Level
+		msg string
+	}{
+		{zap.DebugLevel, "meh"},
+		{zap.InfoLevel, "fwiw"},
+		{zap.ErrorLevel, "hey!"},
+	}
+
+	p := atomic.NewInt32(0)
+
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		i := 0
+		j := p.Inc()
```

Comment:
nit: instead of casting to `int(j)` twice below, you can cast it once here, `j := int(p.Inc())`